docs/index.rst

```diff
@@ -142,10 +142,40 @@ sanity. To achieve this, the codebase is a collection of small reusable
 utilities and abstractions composed together in a few helpful layers.
 
 
+SessionManager
+--------------
+
+:doc:`SessionManager(...) <manager>` is currently urllib3's most abstract manager
```

Comment:
I really like having a first take on it because then my job becomes coming in and simplifying it rather than trying to come up with something from scratch. :)